docs/sources/writing-guide/shortcodes/index.md

```diff
@@ -143,6 +143,43 @@ This shortcode inserts a lists of links to the page's subpages and includes the
 {{</* section withDescriptions="true"*/>}}
 ```
 
+## `docs/reference` shortcode
+
+The `docs/reference` shortcode offers more flexible linking than the Hugo builtin `relref` shortcode.
+Reference links wrapped in the `docs/reference` shortcode can be defined with multiple destinations.
+The destination used in a rendered page derives from matching path prefixes:
+
+```markdown
+{{%/* docs/reference */%}}
+[<LABEL>]: "<PROJECT PATH PREFIX> -> <REFERENCE>"
+...
+{{%/* /docs/reference */%}}
+```
+
+- _`LABEL`_ matches a link label used in a reference-style link (`[LINK TEXT][LABEL])`).
+  For more information about reference-style links, refer to [Daring Fireball: Markdown Syntax Documentation](https://daringfireball.net/projects/markdown/syntax#link).
+- _`PROJECT PATH PREFIX`_ is matched against the page’s URL path, where the first matching `PROJECT PATH PREFIX` is chosen.
+- _`REFERENCE`_ is the parameter that would typically be used in a `relref` shortcode.
+  It can include the syntax `<SOMETHING VERSION>`, which is first looked up in the front matter before falling back to the version inferred from the page's version.
```

Comment:
So this explains what the system is doing with these, but could we instead explain them like what you the writer need to enter. Not exactly as below (I can clean it up if we like this approach, but something similar):

LABEL - The link text
PROJECT PREFIX PATH - The project of the destinations where you want the link to be usable, for example, for grafana, /docs/grafana/, for mimir, /docs/mimir/
REFERENCE - Seems like this is a partial URL (per what we say in our links content currently in the Writer's Toolkit) and not what we use in relrefs?

Reply:
Good point, it perhaps needs tuning to be better understood by TWs.

I really like the solution focused explanation of _`PROJECT PREFIX PATH`_

_`LABEL`_ is not link text but what is called a label in the section about reference-style links from the Markdown author: https://daringfireball.net/projects/markdown/syntax#link. 
Multiple links in a document can use different link text but the same label. That label is resolved once elsewhere in the file, typically at the top or the bottom. Kind of like footnotes.

_`REFERENCE`_ is actually the exact parameter that would be passed to a relref but with the additional version variable substitution. You can actually use absolute paths in relrefs but we have avoided them in versioned documentation because we would need to update the version in each branch of documentation (latest -> v9.5.x once it's no longer latest, etc.). The version variable in the shortcode removes that concern and lets us use those more readable absolute paths.

Reply:
Updated in 237ca9d